* **Natural disasters that occured in all over world: 1900-2021**

In [72]:
import pandas as pd
import numpy as np
import plotly.express as px
import matplotlib.pyplot as plt

In [73]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [74]:
df = pd.read_csv('/content/drive/MyDrive/DATA FILES/1900_2021_DISASTERS.xlsx - emdat data.csv')
df

Year   Seq           Glide Disaster Group Disaster Subgroup  \
0      1900  9002             NaN        Natural    Climatological   
1      1900  9001             NaN        Natural    Climatological   
2      1902    12             NaN        Natural       Geophysical   
3      1902     3             NaN        Natural       Geophysical   
4      1902    10             NaN        Natural       Geophysical   
...     ...   ...             ...            ...               ...   
16121  2021   449  FL-2021-000110        Natural      Hydrological   
16122  2021    75             NaN        Natural      Hydrological   
16123  2021   599  EP-2021-000138        Natural        Biological   
16124  2021    20             NaN        Natural      Hydrological   
16125  2021   481             NaN        Natural      Hydrological   

           Disaster Type Disaster Subtype Disaster Subsubtype   Event Name  \
0                Drought          Drought                 NaN          NaN   
1                Drought          Drought                 NaN          NaN   
2             Earthquake  Ground movement                 NaN          NaN   
3      Volcanic activity         Ash fall                 NaN  Santa Maria   
4      Volcanic activity         Ash fall                 NaN  Santa Maria   
...                  ...              ...                 ...          ...   
16121              Flood              NaN                 NaN          NaN   
16122              Flood              NaN                 NaN          NaN   
16123           Epidemic    Viral disease                 NaN   Meningitis   
16124              Flood              NaN                 NaN          NaN   
16125              Flood              NaN                 NaN          NaN   

                                      Country  ... No Affected No Homeless  \
0                                  Cabo Verde  ...         NaN         NaN   
1                                       India  ...         NaN         NaN   
2                                   Guatemala  ...         NaN         NaN   
3                                   Guatemala  ...         NaN         NaN   
4                                   Guatemala  ...         NaN         NaN   
...                                       ...  ...         ...         ...   
16121                                   Yemen  ...     67980.0         NaN   
16122                            South Africa  ...       400.0         NaN   
16123  Congo (the Democratic Republic of the)  ...         NaN         NaN   
16124                                  Serbia  ...        22.0         NaN   
16125                             South Sudan  ...    500000.0         NaN   

      Total Affected Insured Damages ('000 US$) Total Damages ('000 US$)  \
0                NaN                        NaN                      NaN   
1                NaN                        NaN                      NaN   
2                NaN                        NaN                  25000.0   
3                NaN                        NaN                      NaN   
4                NaN                        NaN                      NaN   
...              ...                        ...                      ...   
16121        67980.0                        NaN                      NaN   
16122          400.0                        NaN                  75000.0   
16123          301.0                        NaN                      NaN   
16124           22.0                        NaN                      NaN   
16125       500000.0                        NaN                      NaN   

            CPI Adm Level                                Admin1 Code  \
0      3.221647       NaN                                        NaN   
1      3.221647       NaN                                        NaN   
2      3.350513       NaN                                        NaN   
3      3.350513       NaN                                        NaN   
4      3.350513    

In [75]:
df.shape

(16126, 45)

* **In this dataset we have imformation about natural disaster that occured in world between year 1900-2021. This data contains 16126 rows and 45 columns.**

In [76]:
df.head(5)

Year   Seq Glide Disaster Group Disaster Subgroup      Disaster Type  \
0  1900  9002   NaN        Natural    Climatological            Drought   
1  1900  9001   NaN        Natural    Climatological            Drought   
2  1902    12   NaN        Natural       Geophysical         Earthquake   
3  1902     3   NaN        Natural       Geophysical  Volcanic activity   
4  1902    10   NaN        Natural       Geophysical  Volcanic activity   

  Disaster Subtype Disaster Subsubtype   Event Name     Country  ...  \
0          Drought                 NaN          NaN  Cabo Verde  ...   
1          Drought                 NaN          NaN       India  ...   
2  Ground movement                 NaN          NaN   Guatemala  ...   
3         Ash fall                 NaN  Santa Maria   Guatemala  ...   
4         Ash fall                 NaN  Santa Maria   Guatemala  ...   

  No Affected No Homeless Total Affected Insured Damages ('000 US$)  \
0         NaN         NaN            NaN                        NaN   
1         NaN         NaN            NaN                        NaN   
2         NaN         NaN            NaN                        NaN   
3         NaN         NaN            NaN                        NaN   
4         NaN         NaN            NaN                        NaN   

  Total Damages ('000 US$)       CPI Adm Level Admin1 Code Admin2 Code  \
0                      NaN  3.221647       NaN         NaN         NaN   
1                      NaN  3.221647       NaN         NaN         NaN   
2                  25000.0  3.350513       NaN         NaN         NaN   
3                      NaN  3.350513       NaN         NaN         NaN   
4                      NaN  3.350513       NaN         NaN         NaN   

  Geo Locations  
0           NaN  
1           NaN  
2           NaN  
3           NaN  
4           NaN  

[5 rows x 45 columns]

In [77]:
df.columns

Index(['Year', 'Seq', 'Glide', 'Disaster Group', 'Disaster Subgroup',
       'Disaster Type', 'Disaster Subtype', 'Disaster Subsubtype',
       'Event Name', 'Country', 'ISO', 'Region', 'Continent', 'Location',
       'Origin', 'Associated Dis', 'Associated Dis2', 'OFDA Response',
       'Appeal', 'Declaration', 'Aid Contribution', 'Dis Mag Value',
       'Dis Mag Scale', 'Latitude', 'Longitude', 'Local Time', 'River Basin',
       'Start Year', 'Start Month', 'Start Day', 'End Year', 'End Month',
       'End Day', 'Total Deaths', 'No Injured', 'No Affected', 'No Homeless',
       'Total Affected', 'Insured Damages ('000 US$)',
       'Total Damages ('000 US$)', 'CPI', 'Adm Level', 'Admin1 Code',
       'Admin2 Code', 'Geo Locations'],
      dtype='object')

In [78]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16126 entries, 0 to 16125
Data columns (total 45 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   Year                        16126 non-null  int64  
 1   Seq                         16126 non-null  int64  
 2   Glide                       1581 non-null   object 
 3   Disaster Group              16126 non-null  object 
 4   Disaster Subgroup           16126 non-null  object 
 5   Disaster Type               16126 non-null  object 
 6   Disaster Subtype            13016 non-null  object 
 7   Disaster Subsubtype         1077 non-null   object 
 8   Event Name                  3861 non-null   object 
 9   Country                     16126 non-null  object 
 10  ISO                         16126 non-null  object 
 11  Region                      16126 non-null  object 
 12  Continent                   16126 non-null  object 
 13  Location                    143

* **drop null columns**

In [79]:
df = df.iloc[:, [0, 3,4,5,6,9, 11,12,13,33,34,35,36,37,39 ]]
df

Year Disaster Group Disaster Subgroup      Disaster Type  \
0      1900        Natural    Climatological            Drought   
1      1900        Natural    Climatological            Drought   
2      1902        Natural       Geophysical         Earthquake   
3      1902        Natural       Geophysical  Volcanic activity   
4      1902        Natural       Geophysical  Volcanic activity   
...     ...            ...               ...                ...   
16121  2021        Natural      Hydrological              Flood   
16122  2021        Natural      Hydrological              Flood   
16123  2021        Natural        Biological           Epidemic   
16124  2021        Natural      Hydrological              Flood   
16125  2021        Natural      Hydrological              Flood   

      Disaster Subtype                                 Country  \
0              Drought                              Cabo Verde   
1              Drought                                   India   
2      Ground movement                               Guatemala   
3             Ash fall                               Guatemala   
4             Ash fall                               Guatemala   
...                ...                                     ...   
16121              NaN                                   Yemen   
16122              NaN                            South Africa   
16123    Viral disease  Congo (the Democratic Republic of the)   
16124              NaN                                  Serbia   
16125              NaN                             South Sudan   

                Region Continent  \
0       Western Africa    Africa   
1        Southern Asia      Asia   
2      Central America  Americas   
3      Central America  Americas   
4      Central America  Americas   
...                ...       ...   
16121     Western Asia      Asia   
16122  Southern Africa    Africa   
16123    Middle Africa    Africa   
16124  Southern Europe    Europe   
16125  Northern Africa    Africa   

                                                Location  Total Deaths  \
0                                            Countrywide       11000.0   
1                                                 Bengal     1250000.0   
2                              Quezaltenango, San Marcos        2000.0   
3                                                    NaN        1000.0   
4                                                    NaN        6000.0   
...                                                  ...           ...   
16121  Dhamar, Amran, Al Mahwit, Marib, Ibb, Sana’a C...          11.0   
16122  Mpumalanga Province, Free State Province and t...          31.0   
16123                                    Tshopo province         131.0   
16124  Zitoradja (Toplicki), Doljevac (Nisavski) , Di...           NaN   
16125  Mayendit County (Unity); Ayod, Fangak Counties...           7.0   

       No Injured  No Affected  No Homeless  Total Affected  \
0             NaN          NaN          NaN             NaN   
1             NaN          NaN          NaN             NaN   
2             NaN          NaN          NaN             NaN   
3             NaN          NaN          NaN             NaN   
4             NaN          NaN          NaN             NaN   
...           ...          ...          ...             ...   
16121         NaN      67980.0          NaN         67980.0   
16122         NaN        400.0          NaN           400.0   
16123       301.0          NaN          NaN           301.0   
16124         NaN         22.0          NaN            22.0   
16125         NaN     500000.0          NaN        500000.0   

       Total Damages ('000 US$)  
0                           NaN  
1                           NaN  
2                       25000.0  
3                           NaN  
4                           NaN  
...                         ...  
16121                       NaN  
16122                   75000.0  
16123                   

In [80]:
df.duplicated().sum()

35

* **Removing Duplicate rows**

In [81]:
df.drop_duplicates(inplace=True)

<ipython-input-81-16cdb8520be8>:1: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



In [82]:
df.dtypes

Year                          int64
Disaster Group               object
Disaster Subgroup            object
Disaster Type                object
Disaster Subtype             object
Country                      object
Region                       object
Continent                    object
Location                     object
Total Deaths                float64
No Injured                  float64
No Affected                 float64
No Homeless                 float64
Total Affected              float64
Total Damages ('000 US$)    float64
dtype: object

In [83]:
px.area(df.groupby('Year').count(),df.groupby('Year').count().index, 'Disaster Group', title= 'Disaster Count Per Year', template='presentation', width=800).update_layout(xaxis_title = 'Year', yaxis_title = 'Disaster Count')

* **532 disasters occured in year-2002**

In [84]:
px.line(df.groupby('Year').sum(),df.groupby('Year').count().index, 'Total Affected', title= 'Disaster Affect by Year', template='presentation', width=800).update_layout(xaxis_title = 'Year', yaxis_title = 'No. of Affected People')

<ipython-input-84-331903dcc949>:1: FutureWarning:

The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.



* **658 M people are affected in 2002 disasters**

In [85]:
px.pie(df, 'Continent',labels='Continent', title= 'Disasters by Continent', template='presentation', width=500).update_traces(textinfo='label+percent+value', textposition='inside')

* **40% of disasters comes in Asia**

In [86]:
df['Disaster Type'].value_counts()

Flood                    5549
Storm                    4476
Earthquake               1542
Epidemic                 1501
Landslide                 776
Drought                   770
Extreme temperature       603
Wildfire                  460
Volcanic activity         265
Insect infestation         96
Mass movement (dry)        48
Glacial lake outburst       2
Fog                         1
Impact                      1
Animal accident             1
Name: Disaster Type, dtype: int64

In [87]:
px.bar(df.groupby('Disaster Type').count() , df.groupby('Disaster Type').count().index, 'Disaster Group', text='Disaster Group', title= 'No. of disasters per type of disaster', template='presentation', width=900).update_layout(xaxis_title = None, yaxis_title = None)

* **Flood and Storm both happens a lot.**

* We have null value in totaldeaths column but we can't remove or fill that column and for some specific task we have to replace null value with 0.

In [88]:
df['Total Deaths'].replace(np.NaN, 0, inplace = True)
#fillna()

<ipython-input-88-3c1512f3c6ef>:1: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



In [89]:
px.scatter(df.sort_values('Total Deaths', ascending = False).head(30), y ='Total Deaths', x = 'Country', color='Continent', hover_data=df.columns, size='Total Deaths', size_max=30, title = 'Top Disasters With Highest No. Deaths', template='presentation', width=900)

* **In 1931 china floods 3700000 peoples are died.**

In [90]:
px.pie(df, names='Disaster Subgroup', values='Total Deaths', facet_col='Continent', hole=.5, title='Total deaths by Disaster subgroup', template='presentation')

1. **60% of Total deaths of disasters in Africa are due to Climatological**
2. **60% of Total deaths of disasters in Americas are due to Geophysical**
3. **60% of Total deaths of disasters in Europe are due to Biological**

In [91]:
df

Year Disaster Group Disaster Subgroup      Disaster Type  \
0      1900        Natural    Climatological            Drought   
1      1900        Natural    Climatological            Drought   
2      1902        Natural       Geophysical         Earthquake   
3      1902        Natural       Geophysical  Volcanic activity   
4      1902        Natural       Geophysical  Volcanic activity   
...     ...            ...               ...                ...   
16121  2021        Natural      Hydrological              Flood   
16122  2021        Natural      Hydrological              Flood   
16123  2021        Natural        Biological           Epidemic   
16124  2021        Natural      Hydrological              Flood   
16125  2021        Natural      Hydrological              Flood   

      Disaster Subtype                                 Country  \
0              Drought                              Cabo Verde   
1              Drought                                   India   
2      Ground movement                               Guatemala   
3             Ash fall                               Guatemala   
4             Ash fall                               Guatemala   
...                ...                                     ...   
16121              NaN                                   Yemen   
16122              NaN                            South Africa   
16123    Viral disease  Congo (the Democratic Republic of the)   
16124              NaN                                  Serbia   
16125              NaN                             South Sudan   

                Region Continent  \
0       Western Africa    Africa   
1        Southern Asia      Asia   
2      Central America  Americas   
3      Central America  Americas   
4      Central America  Americas   
...                ...       ...   
16121     Western Asia      Asia   
16122  Southern Africa    Africa   
16123    Middle Africa    Africa   
16124  Southern Europe    Europe   
16125  Northern Africa    Africa   

                                                Location  Total Deaths  \
0                                            Countrywide       11000.0   
1                                                 Bengal     1250000.0   
2                              Quezaltenango, San Marcos        2000.0   
3                                                    NaN        1000.0   
4                                                    NaN        6000.0   
...                                                  ...           ...   
16121  Dhamar, Amran, Al Mahwit, Marib, Ibb, Sana’a C...          11.0   
16122  Mpumalanga Province, Free State Province and t...          31.0   
16123                                    Tshopo province         131.0   
16124  Zitoradja (Toplicki), Doljevac (Nisavski) , Di...           0.0   
16125  Mayendit County (Unity); Ayod, Fangak Counties...           7.0   

       No Injured  No Affected  No Homeless  Total Affected  \
0             NaN          NaN          NaN             NaN   
1             NaN          NaN          NaN             NaN   
2             NaN          NaN          NaN             NaN   
3             NaN          NaN          NaN             NaN   
4             NaN          NaN          NaN             NaN   
...           ...          ...          ...             ...   
16121         NaN      67980.0          NaN         67980.0   
16122         NaN        400.0          NaN           400.0   
16123       301.0          NaN          NaN           301.0   
16124         NaN         22.0          NaN            22.0   
16125         NaN     500000.0          NaN        500000.0   

       Total Damages ('000 US$)  
0                           NaN  
1                           NaN  
2                       25000.0  
3                           NaN  
4                           NaN  
...                         ...  
16121                       NaN  
16122                   75000.0  
16123                   

In [92]:
px.bar(df[df['Disaster Type'] == 'Epidemic'].groupby('Country').count().sort_values('Year', ascending = True).reset_index().tail(20), y= 'Country', x='Year' , orientation = 'h', color= 'Total Affected', title= 'Total number of Epidemic per Country').update_layout(xaxis_title = "No.of Epidemic happens in country")

* **Epidemic has happened 70 times in India**

In [93]:
px.funnel(df.groupby('Country').sum().sort_values("Total Damages ('000 US$)", ascending = False).reset_index().head(10),y= 'Country', x="Total Damages ('000 US$)", title = 'Total Damages by Country')

<ipython-input-93-ce3cd62fbcfa>:1: FutureWarning:

The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.



* **USA has lost the most money in disaster**

In [94]:
px.choropleth(df.groupby('Country').sum().reset_index(), locationmode='country names', locations='Country', color = "Total Deaths", template='presentation')

<ipython-input-94-b02e0104e2a2>:1: FutureWarning:

The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.



In [95]:
px.treemap(df[df['Total Deaths'] > 0], path= [px.Constant('World'), 'Continent','Region','Country'], values= 'Total Deaths', color='Total Deaths', title='Treemap via Number of Deaths in Disasters',template='presentation')